File: eureka/ios/nativeLib/nativeLib/Models/Responses/CalibrationResponse.swift
//
//  CalibrationResponse.swift
//  eureka
//
//  Created by work on 19/02/21.
//

import Foundation

public class CalibrationResponse : ResponseBase {
  private var _userId:String = ""
  private var _authId:String = ""

  internal override init(pResponseBase: ResponseBase) {
    super.init(pResponseBase:pResponseBase)
  }
  
  override init(pErrorCode:ResultCodeEnum) {
    super.init(pErrorCode: pErrorCode)
  }

  public override init(pErrorCode:ResultCodeEnum, pMessage:String) {
    super.init(pErrorCode: pErrorCode, pMessage: pMessage)
  }

/*
  public CalibrationResponse(String pMessage) {
      super(pMessage);
  }
*/

  internal override init(pSimuId:String, pSimuMessage:String) {
    super.init(pSimuId: pSimuId, pSimuMessage: pSimuMessage)
  }

  public override func getResponseStr() -> String {
    do {
      var mResultMap:[String:Any] = [:]

      var result:[String:Any] = getResponseMap()
      var mData:[String:String] = [:]
      mData["userId"] = _userId
      mData["AuthenticationId"] = _authId
      result["data"] = mData

      mResultMap["result"] = result
      let jsonData = try JSONSerialization.data(withJSONObject: mResultMap, options: [])
      let jsonString = String(data: jsonData, encoding: String.Encoding.ascii)!

      return jsonString
    } catch _ {
      return ""
    }
  }

  internal func getUserId() -> String {
    return _userId
  }

  public func setUserId(pUserId:String) {
    _userId = pUserId
  }

  internal func getAuthId() -> String {
    return _authId
  }

  public func setAuthId(pAuthId:String) {
    _authId = pAuthId
  }
}

File: eureka/android/library/src/main/java/com/lifeplus/Pojo/Responses/CalibrationResponse.java
package com.lifeplus.Pojo.Responses;

import com.google.gson.Gson;
import com.lifeplus.Pojo.Enum.ResultCodeEnum;

import java.util.HashMap;

public class CalibrationResponse extends ResponseBase {
    private String _userId = "";
    private String _authId = "";

    public CalibrationResponse(ResultCodeEnum pErrorCode) {
        super(pErrorCode);
    }

    public CalibrationResponse(ResultCodeEnum pErrorCode, String pMessage) {
        super(pErrorCode, pMessage);
    }

/*
    public CalibrationResponse(String pMessage) {
        super(pMessage);
    }
*/

    public CalibrationResponse(String pSimuId, String pSimuMessage) {
        super(pSimuId, pSimuMessage);
    }

    @Override
    public String getResponseStr() {
        HashMap<String, Object> mResultMap = new HashMap<>();

        HashMap<String, Object> result = getResponseMap();
        HashMap<String, String> mData = new HashMap<>();
        mData.put( "userId", _userId );
        mData.put( "AuthenticationId", _authId );
        result.put( "data", mData );

        mResultMap.put( "result",  result);
        return new Gson().toJson( mResultMap );
    }

    public String getUserId() {
        return _userId;
    }

    public void setUserId(String pUserId) {
        _userId = pUserId;
    }

    public String getAuthId() {
        return _authId;
    }

    public void setAuthId(String pAuthId) {
        _authId = pAuthId;
    }
}
